bridge/client/app/_views/ktb-settings-view/ktb-integration-view/ktb-integration-view.component.ts

```diff
@@ -165,24 +109,53 @@ export class KtbIntegrationViewComponent implements OnInit, OnDestroy {
     return (registration.hasSubscriptions(projectName) ? undefined : template) as TemplateRef<unknown>;
   }
 
-  private compare(a: string, b: string, isAsc: boolean): number {
-    const result = a.localeCompare(b);
-    if (result !== 0 && !isAsc) {
-      return -result;
-    }
-    return result;
+  public getSubscriptions(uniformRegistration: UniformRegistration, projectName: string): string[] {
+    return uniformRegistration.subscriptions
+      .filter((s) => s.hasProject(projectName, true))
+      .map((s) => s.formattedEvent);
+  }
+
+  public toUniformRegistration(item: unknown): UniformRegistration {
+    return <UniformRegistration>item;
   }
 
-  public formatSubscriptions(uniformRegistration: UniformRegistration, projectName: string): string | undefined {
-    const subscriptions = uniformRegistration.subscriptions.reduce(
-      (accSubscriptions: string[], subscription: UniformSubscription) => {
-        if (subscription.hasProject(projectName, true)) {
-          accSubscriptions.push(subscription.formattedEvent);
-        }
-        return accSubscriptions;
-      },
-      []
+  private loadLogs(uniformRegistrationId: string): Observable<UniformRegistrationLog[]> {
+    return this.dataService.getUniformRegistrationLogs(uniformRegistrationId).pipe(
+      tap((logs) => this.setUniformDate(uniformRegistrationId, logs)),
+      finalize(() => (this.isLoadingLogs = false))
     );
-    return subscriptions.length !== 0 ? subscriptions.join('<br/>') : undefined;
   }
+
+  private setUniformDate(uniformRegistrationId: string, logs: UniformRegistrationLog[]): void {
+    this.dataService.setUniformDate(uniformRegistrationId, logs[0]?.time);
+  }
+}
+
+export function sortRegistrations(
+  registrations: UniformRegistration[],
+  column: string,
+  ascending: boolean
+): UniformRegistration[] {
+  return [...registrations].sort((a: UniformRegistration, b: UniformRegistration) => {
+    const sortBy = sortConfig[column];
+    const sortResult = sortBy ? compare(sortBy(a), sortBy(b), ascending) : 0;
+    return sortResult || compare(a.name, b.name, ascending);
+  });
+}
+
+export function sortLogs(logs: UniformRegistrationLog[]): UniformRegistrationLog[] {
+  return logs.sort((a, b) => {
+    if (a.time.valueOf() > b.time.valueOf()) {
+      return -1;
+    } else if (a.time.valueOf() < b.time.valueOf()) {
+      return 1;
+    }
+    return 0;
```

Comment:
Could either be written as `a.time - b.time` or before defining two consts to prevent executing `valueOf` twice

Reply:
Fixed this by actually comparing dates

Reply:
See unit test